packages/observable/src/observable.ts

```diff
@@ -10,63 +10,171 @@ import * as dependencyDetection from './dependencyDetection'
 import { deferUpdates } from './defer'
 import { subscribable, defaultEvent, LATEST_VALUE } from './subscribable'
 import { valuesArePrimitiveAndEqual } from './extenders'
+import type { Subscribable, MaybeSubscribable } from './subscribable'
 
-export function observable (initialValue) {
-  function Observable () {
+//#region Observable
+
+/**
+ * Represents a value that can be either a plain value or an observable.
+ */
+export type MaybeObservable<T = any> = T | Observable<T>;
+
+/**
+ * Defines the functions available on an observable.
+ */
+export interface ObservableFunctions<T = any> extends Subscribable<T> {
+  /**
+   * Compares two values for equality.
+   * @param a The first value.
+   * @param b The second value.
+   * @returns True if the values are equal, otherwise false.
+   */
+  equalityComparer(a: T, b: T): boolean;
+
+  /**
+   * Returns the current value of the observable without creating a dependency.
+   * @returns The current value.
+   */
+  peek(): T;
+
+  /**
+   * Notifies subscribers that the value has changed.
+   */
+  valueHasMutated(): void;
+
+  /**
+   * Notifies subscribers that the value is about to change.
+   */
+  valueWillMutate(): void;
+
+  /**
+   * Modifies the value of the observable using a function.
+   * @param fn The function to modify the value.
+   * @param peek Whether to use the current value without creating a dependency.
+   * @returns The modified observable.
+   */
+  modify(fn, peek? : Boolean): Observable
+}
+
+/**
+ * Represents an observable value.
+ */
+export interface Observable<T = any> extends ObservableFunctions<T> {
+  /**
+   * The latest value of the observable.
+   */
+  [LATEST_VALUE]: any;
+
+  /**
+   * Gets the current value of the observable.
+   * @returns The current value.
+   */
+  (): T;
+
+  /**
+   * Sets the value of the observable.
+   * @param value The new value.
+   * @returns The observable.
+   */
+  (value: T): any;
+}
+
+//#endregion Observable  
+
+/**
+ * Creates an observable object.
+ * @param initialValue The initial value of the observable.
+ * @returns The observable object.
+ */
+export function observable(initialValue?: any): Observable {
```

Comment:
should be ?
```ts
export function observable<T=any> (initialValue: T): Observable<T> {
```

Reply:
Should be, but we have 76 foolish test cases or rather known KO behavior.

![image](https://github.com/user-attachments/assets/5f02f80e-17be-4a26-bb5b-bc30158fff28)

I have change it "transitionally" to
```typescript
export function observable<T=any>(initialValue?: T): Observable
```